### PR TITLE
gh-90535: Warn of deprecation of intervals other than 1 with `when='MIDNIGHT'`

### PR DESCRIPTION
**Pull request by vsajip**
* Issue: gh-90535

### CODE REVIEW

File: Lib/test/test_logging.py [hunk 1]
```diff
@@ -5755,6 +5755,16 @@ def test_compute_rollover_daily_attime(self):
         finally:
             rh.close()
 
+    def test_midnight_interval(self):
+        with warnings.catch_warnings(record=True) as w:
+            rh = logging.handlers.TimedRotatingFileHandler(
+                self.fn, encoding="utf-8", when='MIDNIGHT', interval=2, backupCount=0,
+                utc=True)
+            rh.close()
+        if w:
```
Comment (brettcannon) [inline Lib/test/test_logging.py:6037]: Do you want the test to pass if no warning is raised? If not then in the above I would force deprecations on as exceptions via `warnings.simplefilter("error")`.

Comment (vsajip) [reply Lib/test/test_logging.py:6037]: Good point. Thanks for the review, Brett.

File: Lib/logging/handlers.py [hunk 2]
```diff
@@ -28,6 +28,7 @@
 import queue
 import threading
 import copy
+import warnings
```
Comment (serhiy-storchaka) [inline Lib/logging/handlers.py:31]: It is better to defer the `warnings` module import until it is needed (which never happen in normal case).

File: Lib/test/test_logging.py (hunk 1 above)
Comment (serhiy-storchaka) [reply Lib/test/test_logging.py:6037]: Why not use `assertWarns()`?

**Review by serhiy-storchaka — COMMENTED** [review]
I do not think that this change is needed. What if you want to roll over at 4:00 every second night? What are the alternatives?

**Comment by serhiy-storchaka** [comment]
#116220 is an alternate solution, which fixes support of intervals>1 instead of deprecating it for when='MIDNIGHT' and when='Wx'.

**Comment by serhiy-storchaka** [comment]
I guess it can be closed now.

**Pull request closed without merging**